### ISSUE TITLE
Change on the behavior of markdown links with the release of v0.12

### ISSUE THREAD

**Issue by danielarmengolaltayo** [foambubble/foam#551]
_The context for this issue matches with #547 (the change appeared with v0.12) and seems related with #524. I decided to open a new issue because the behavior I want to highlight is another one._ 

Before 0.12, I could navigate the wikilinks inside the markdown links: `[some text]([[wikilink]])`

Before:

<img width="333" alt="Captura de pantalla 2021-03-30 a las 22 32 26" src="https://user-images.githubusercontent.com/7153778/113053033-236fe100-91a8-11eb-9427-750b98999de1.png">

v0.12

<img width="639" alt="Captura de pantalla 2021-03-30 a las 22 33 50" src="https://user-images.githubusercontent.com/7153778/113053058-2d91df80-91a8-11eb-8c10-4841236d522f.png">

_Interesting to see in the last screenshot how the preview works, although it is behaving as the note does not exist._

**Comment by danielarmengolaltayo** [comment]
I've just found out that the same is happening with markdown links pointing to other files that are not markdown. 

<img width="536" alt="Captura de pantalla 2021-03-31 a las 17 55 50" src="https://user-images.githubusercontent.com/7153778/113174122-59fe3800-924a-11eb-83ba-5d2c61eaabc3.png">

**Comment by riccardoferretti** [comment]
Hey Daniel, I am trying to better understand the issue, as there are a few things in it:

1.
I am not sure I understand the syntax `[label]([[wikilink]])`. 
With wikilinks aliases are generally achieved by using `[[wikilink|label]]` syntax, which we don't yet (but will soon) support.
Not sure whether the `[label]([[wikilink]])` is standard markdown and we should support it.
I agree the preview behavior is odd, I believe it's coming from vscode-memo as we don't do hover preview for links yet.
If that's the case, the odd behavior is due to the combo of foam + vscode-memo

2.
will make decorations optional from 0.12.1, is this the problem you are describing?

3.
that's a good point that links to non-md files that exist should work, foam atm doesn't take into account attachments, that's the reason for the issue

Can you tell me what this issue is specifically about? so we can better categorize it

**Comment by danielarmengolaltayo** [comment]
1 is the main issue, 2 is not the problem I am trying to describe here (there are other issues about it) and 3 is a secondary but related issue I recognized later. Let me develop 1 and 3:

1. You are right, what I am trying to achieve is the aliases' behavior, but because Foam does not support it (yet), I found this workaround which works fine with my published site. Regardless of contemplating my specific use case, the thing here is that Foam v0.12 prioritizes first the markdown link syntax before the wikilink syntax, so the wikilink is not accessible because it is inside a markdown link which it has a new behavior attached to its syntax. And I don't have vscode-memo (aka markdown-memo) installed.

3. For example, if I have an image I can point to it and open it directly inside vscode. Using the same technique I can point to a folder, what translates into finding the folder and selecting it in the explorer panel, my strategy to connect Foam with my file system, a feature I heavily use and one of the key components I think Foam could explore in the future, but that is something for another conversation. I can not recognize right now from which extension this behavior is coming, but Foam is overriding it.

Hope it is more clear now :)
